identity/client/src/utility/api/users/index.ts

```diff
@@ -1,10 +1,11 @@
-import { ApiDefinition, apiDelete, apiGet, apiPost, apiPut } from "../request";
+import { ApiDefinition, apiDelete, apiPost, apiPut } from "../request";
 import { SearchResponse } from "src/utility/api";
+import { EntityData } from "src/components/entityList/EntityList";
 
 export const USERS_ENDPOINT = "/users";
 
-export type User = {
-  id: string;
+export type User = EntityData & {
+  key: number;
```

Comment:
This makes `id` an optional property of User (because it's optional in the EntityData type). And it makes it necessary to use the non-null assertion operator (`!`) a lot on `user.id!`. 

Is this intentional or can `id` EntityData be made non-optional?

Reply:
I think id is optional because when entity is not created yet (stored in backend) the value is null.

Reply:
Okay, but in the places you use `user.id!` it is guaranteed that it's not null? Otherwise there might be unexpected behavior which can't be caught by typescript.

Reply:
you are right and as I understand from the flow it's available in this place as it loads a detail of an existing user.